summarize/src/main.rs

```diff
@@ -25,17 +43,96 @@ struct Opt {
     percent_above: f64,
 }
 
-fn main() -> Result<(), Box<dyn Error>> {
-    let opt = Opt::from_args();
+#[derive(StructOpt, Debug)]
+enum Opt {
+    #[structopt(name = "diff")]
+    Diff(DiffOpt),
+
+    /// Processes trace files and produces a summary
+    #[structopt(name = "summarize")]
+    Summarize(SummarizeOpt),
+}
+
+fn process_results(file: &PathBuf) -> Result<Results, Box<dyn Error>> {
+    if file.ends_with("json") {
+        let reader = File::open(&file)?;
+
+        let results: Results = serde_json::from_reader(reader)?;
```

Comment:
I did rather go for wrapping it in `BufReader`, so that a file of say 4 gb doesnt get loaded into memory at once. The parsed result is probably smaller, as json contains a lot of special chars, which are not part of the actual parsed content. The [`with_capacity`](https://doc.rust-lang.org/stable/std/io/struct.BufReader.html#method.with_capacity) method allows building with a large capacity, like 128mb. Using a `BufReader` can also reduce latency, as parsing can already start while the file isnt fully read yet.